Realm/RLMQueryUtil.mm

```diff
@@ -207,10 +207,18 @@ void add_string_constraint_to_query(realm::Query &query,
     }
 }
 
+void add_string_constraint_to_query(realm::Query &query,
+                                    NSPredicateOperatorType operatorType,
+                                    NSComparisonPredicateOptions predicateOptions,
+                                    Columns<String> &&column,
+                                    id value) {
+    add_string_constraint_to_query(query, operatorType, predicateOptions, std::forward<Columns<String>>(column), RLMStringDataWithNSString(value));
+}
+
 void add_string_constraint_to_query(realm::Query& query,
                                     NSPredicateOperatorType operatorType,
                                     NSComparisonPredicateOptions predicateOptions,
-                                    NSString *value,
+                                    id value,
```

Comment:
It's not ideal, but it's not a big departure from what happens prior to this change. The code was already implicitly converting an `id` to `NSString *`. It now just happens at the point `RLMStringDataWithNSString` is called by `add_string_constraint_to_query`, rather than when `add_string_constraint_to_query` is called.